native_field/src/columns.rs

```diff
@@ -5,20 +5,26 @@ use valida_machine::Word;
 use valida_util::indices_arr;
 
 #[derive(AlignedBorrow, Default)]
-pub struct Mersenne31Cols<T> {
+pub struct NativeFieldCols<T> {
     pub input_1: Word<T>,
     pub input_2: Word<T>,
 
     /// Witnessed output
     pub output: Word<T>,
 
+    pub is_add: T,
+    pub is_sub: T,
+    pub is_mul: T,
+
     pub opcode: T,
```

Comment:
I think we could potentially replace this with `is_add * ADD_OPCODE + is_sub * SUB_OPCODE + is_mul * MUL_OPCODE`? Since it's degree 1

Reply:
I believe we can't replace this (or `is_real`) yet for this same reason: https://github.com/valida-xyz/valida/pull/14/files#r1268763247

I'll create a new PR to address this.

Reply:
I added some new constructors like `VirtualPairCol::new_main` which I think we can use here.

(There is still the remaining limitation that we can't refer to "next" columns, affecting the output chip)